src/deploy/table_invitation_policy.sql

```diff
@@ -14,62 +16,99 @@ GRANT INSERT, DELETE ON TABLE maevsi.invitation TO maevsi_account;
+	    SELECT c.id
+	    FROM maevsi.contact c
+	      JOIN maevsi.account_block b ON c.account_id = b.author_account_id AND c.author_account_id = b.blocked_account_id
+	    WHERE c.account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
     )
   )
-  OR  event_id IN (SELECT maevsi.events_organized())
+  OR (
+	  event_id IN (SELECT maevsi.events_organized())
+	AND
+	  contact_id NOT IN (
+	    SELECT c.id
+	    FROM maevsi.contact c
+	      JOIN maevsi.account_block b
+	      -- contact authored by a blocked account OR referring to a blocked account
+	      ON c.author_account_id = b.blocked_account_id OR c.account_id = b.blocked_account_id
+	    WHERE b.author_account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
+	  )
+  )
 );
 
 -- Only allow inserts for invitations to events organized by oneself.
 -- Only allow inserts for invitations to events for which the maximum invitee count is not yet reached.
--- Only allow inserts for invitations issued to a contact that was created by oneself.
+-- Only allow inserts for invitations issued to a contact that was created by oneself
+-- Do not allow inserts for invitations issued to a contact referring a blocked account
 CREATE POLICY invitation_insert ON maevsi.invitation FOR INSERT WITH CHECK (
-      event_id IN (SELECT maevsi.events_organized())
+  event_id IN (SELECT maevsi.events_organized())
   AND (
     maevsi.event_invitee_count_maximum(event_id) IS NULL
     OR
     maevsi.event_invitee_count_maximum(event_id) > maevsi.invitee_count(event_id)
   )
   AND
-  (
-    NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID IS NOT NULL
-    AND
     contact_id IN (
       SELECT id
       FROM maevsi.contact
-      WHERE contact.author_account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
-    )
-  )
+      WHERE author_account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
+
+	    EXCEPT
+
+	    SELECT c.id
+	    FROM maevsi.contact c
+	      JOIN maevsi.account_block b ON c.account_id = b.blocked_account_id and c.author_account_id = b.author_account_id
+	    WHERE c.author_account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
+	  )
 );
 
 -- Only allow updates to invitations issued to oneself through invitation claims.
--- Only allow updates to invitations issued to oneself through the account.
--- Only allow updates to invitations to events organized by oneself.
+-- Only allow updates to invitations issued to oneself through the account, but not invitations auhored by a blocked account
+-- Only allow updates to invitations to events organized by oneself, but not invitations issued to a blocked account or issued by a blocked account
 CREATE POLICY invitation_update ON maevsi.invitation FOR UPDATE USING (
   id = ANY (maevsi.invitation_claim_array())
   OR
   (
-    NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID IS NOT NULL
-    AND
```

Comment:
I think the same as outlined above applies here: anonymous users could update invitations for contacts that have no account assigned.

Reply:
Answer from above applies here as well